### PR TITLE
Add folding argument to pwd()

### PR DESCRIPTION
**Pull request by hooman96**
This resolves issue #2

### CODE REVIEW

File: SwiftShell.playground/Contents.swift [hunk 1]
```diff
@@ -8,4 +8,10 @@
 
 import SwiftShell
 
-echo("Hello, SwiftShell!")
```
Comment (JustinJiaDev) [inline SwiftShell.playground/Contents.swift:11]: Exclude changes to SwiftShell.playground. This file should keep as it is and only be changed by actual users.

File: SwiftShell/Git.swift [hunk 2]
```diff
@@ -47,5 +48,16 @@ public enum Git {
     public static func commit(_ message: String, options: CommitOption...) -> String {
         return git(command: .commit, arguments: ["-m", message] + options.map { $0.rawValue })
     }
-
+    
+    public enum CheckOutOption: String {
```
Comment (JustinJiaDev) [inline SwiftShell/Git.swift:52]: Remove commented code and checkOutOption since they are not finished.

File: SwiftShell/Print.swift [hunk 3]
```diff
@@ -11,7 +11,7 @@ import Foundation
 func print(command: String, with arguments: [String]? = nil) {
     var output = "SwiftShell $ \(command)"
     if let arguments = arguments {
-        output += arguments.reduce("", { $0 + " " + $1 })
+        output += arguments.reduce("", combine: { $0 + " " + $1 })
```
Comment (JustinJiaDev) [inline SwiftShell/Print.swift]: Revert this change. You should update to Xcode 8 Beta 4. Swift 3 beta 4 removed combine label.

File: SwiftShell/PrintWorkingDirectory.swift [hunk 4]
```diff
@@ -8,11 +8,19 @@
...
 
-public func pwd() -> String {
+public func pwd(folding: Bool = false) -> String {
     print(command: "pwd")
     var buffer: [Int8] = Array(repeating: 0, count: Int(PATH_MAX))
     getcwd(&buffer, buffer.count)
-    let output = String(cString: buffer) ?? ""
+    var output = String(cString: buffer) ?? ""
+    
+    let outputArr = output.characters.split{$0 == "/"}.map(String.init)
```
Comment (JustinJiaDev) [inline SwiftShell/PrintWorkingDirectory.swift]: - Change `String.init` to `""`.
- Rename `outputArr` to `pathComponents`.
- Add spaces before and after first curly bracket. Add space before second curly bracket.

e.g.

``` swift
let pathComponents = output.character.split { $0 == "/" }.map("")
```

File: SwiftShell/PrintWorkingDirectory.swift [hunk 5]
```diff
@@ -8,11 +8,19 @@
...
+
+    if(output.characters.count > 15 && folding == true) {
```
Comment (JustinJiaDev) [inline SwiftShell/PrintWorkingDirectory.swift]: - Add space after if.
- Remove brackets.
- Don't force unwrap optionals. Use if let instead.
- We can actually remove 15 character constraint. Don't add surprising behavior without documentation. In the future, we can change folding to a enum or add another argument (e.g. characterLimit:).

e.g.

``` swift
if let firstPathComponents = pathComponents.first, let lastPathComponents = pathComponents.last, folding == true {
```

File: SwiftShell/PrintWorkingDirectory.swift [hunk 6]
```diff
@@ -8,11 +8,19 @@
...
+        output = outputArr.first! + "/.../" + outputArr.last!
```
Comment (JustinJiaDev) [inline SwiftShell/PrintWorkingDirectory.swift]: Now you can use unwrapped optionals.

e.g.

``` swift
output = firstPathComponents + "/.../" + lastPathComponents
```

File: SwiftShell/PrintWorkingDirectory.swift [hunk 7]
```diff
@@ -8,11 +8,19 @@
...
+        print(output)
```
Comment (JustinJiaDev) [inline SwiftShell/PrintWorkingDirectory.swift]: Remove this duplicated print.

File: SwiftShell/String.swift [hunk 8]
```diff
@@ -11,7 +11,7 @@ import Foundation
 extension String {
     
     func trimmedLastEndOfLine() -> String {
-        guard let lastCharacter = self.characters.last, lastCharacter == "\n" else { return self }
```
Comment (JustinJiaDev) [inline SwiftShell/String.swift:14]: Revert this change. Syntax is changed in Swift 3 beta 4.

File: SwiftShell/PrintWorkingDirectory.swift (hunk 4 above)
Comment (hooman96) [inline SwiftShell/PrintWorkingDirectory.swift]: what guide lines do you follow for swift code style?

File: SwiftShell/PrintWorkingDirectory.swift (hunk 6 above)
Comment (hooman96) [inline SwiftShell/PrintWorkingDirectory.swift]: safer to use unwrapped optionals since they don't change?

Comment (hooman96) [inline SwiftShell/PrintWorkingDirectory.swift]: got an error using "" 

Cannot convert value of type 'String' to expected argument type '@noescape (SubSequence) throws -> _'

**Comment by JustinJiaDev** [comment]
Thanks! Made a few changes, like [this](https://github.com/JustinJiaDev/SwiftShell/commit/d58e5d07490461c2c16880c133ad372c768c22a9) and [this](https://github.com/JustinJiaDev/SwiftShell/commit/8fd0dfa70d10890b1c1d59369ef6c6ea7687bd4b). You can check them out if you wish.

One suggestion: in the future you can create different branches e.g. (branch `add-folding-to-pwd` or branch `rename-open-directory`...).  A few advantages:
1. You can safely delete the branch after pull request is merged since you are only going to use it once.
2. You can create a new branch from master next time. Right now you need to merge the latest changes from master to your develop branch, and you need to handle merge conflicts if you forget to do so.
3. You only need to focus on your current work. Don't change unnecessary files so merge conflict is less likely to happen. If you want to change something else, you can always create another pull request with another branch.
4. Pull request can be more descriptive since it will only contain your current work.
5. You can increase the chance of passing the code review without revision since less code will be reviewed in one pull request.

BTW, I deleted you branch in case you forget to merge master changes.

**Comment by JustinJiaDev** [comment]
@hooman96 SwiftShell should follow this guideline: https://swift.org/documentation/api-design-guidelines/

File: SwiftShell/PrintWorkingDirectory.swift (hunk 6 above)
Comment (JustinJiaDev) [inline SwiftShell/PrintWorkingDirectory.swift]: If you force-unwrap an optional by writing code like `variable!`, the program will crash if the variable is `nil`. Never use `!` unless it's absolutely necessary. Use `if let` and question mark instead.

Yeah you should use `String.init`. I thought `map` takes in an instance variable.

**Comment by hooman96** [comment]
i see. I thought usually each developer has a branch and master has the most updated production code. And we just close the PR with every updates but will keep our branches and merge conflicts to master branch when code is stable. but we can just push to master and close the PRs and branches.

I will follow the guide lines then. Is there automated tool to run at the end to clean up the code, indentions and comments? similar to pep8 in python 
https://pypi.python.org/pypi/pep8

i found this: https://github.com/realm/SwiftLint

**Comment by JustinJiaDev** [comment]
It is preferred to separate work by "feature" instead of "developer". It's very likely that two developers will work on the same feature for some period of time.

`SwiftLint` isn't very flexible yet. Can't rely on it to do all the work.
